internal/controllers/gateway/httproute_controller.go

```diff
@@ -445,14 +447,12 @@ func (r *HTTPRouteReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	// we can update the object status to indicate that it's now properly linked
 	// to the configured Gateways.
 	debug(log, httproute, "Ensuring status contains Gateway associations")
-	statusUpdated, err := r.ensureGatewayReferenceStatusAdded(ctx, httproute, gateways...)
+	shouldRequeue, err := r.ensureGatewayReferenceStatusAdded(ctx, httproute, gateways...)
 	if err != nil {
 		// don't proceed until the statuses can be updated appropriately
 		return ctrl.Result{}, err
 	}
-	if statusUpdated {
-		// if the status was updated it will trigger a follow-up reconciliation
-		// so we don't need to do anything further here.
+	if shouldRequeue {
```

Comment:
I think, it may be non-obvious for someone why it will requeue when empty `ctrl.Result{}` is returned

Reply:
Agreed, I've refactored it and added back the `updated` return value. 3ad34d5a2072e56e9fe9618c15ce2abfafa8daaf

PTAL